src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -201,51 +257,119 @@ else if (pos != null)
+        {
+            @Override
+            public int getElementCount()
+            {
+                return sections.size();
+            }
+
+            @Override
+            public String getLabel(final int index)
+            {
+                final String name = sections.get(index);
+                if (Structures.SCHEMATICS_SCAN.equals(name))
+                {
+                    return LanguageHandler.format("com.minecolonies.coremod.gui.buildtool.scans");
+                }
+                else if (Structures.SCHEMATICS_PREFIX.equals(name))
+                {
+                    return LanguageHandler.format("com.minecolonies.coremod.gui.buildtool.decorations");
+                }
+                //should be a hut
+                return LanguageHandler.format("tile.minecolonies.blockHut" + name + ".name");
+            }
+        });
     }
 
     /**
-     * Move the schmatic up.
+     * Initialise the previous/next and drop down list for style.
      */
-    private static void moveUpClicked()
+    private void initStyleNavigation()
     {
-        Settings.instance.moveTo(new BlockPos(0, 1, 0));
+        registerButton(BUTTON_PREVIOUS_STYLE_ID, this::previousStyle);
+        registerButton(BUTTON_NEXT_STYLE_ID, this::nextStyle);
+        stylesDropDownList = findPaneOfTypeByID(DROPDOWN_STYLE_ID, DropDownList.class);
+        stylesDropDownList.setHandler((DropDownList.Handler)this);
+        stylesDropDownList.setDataProvider(new DropDownList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return styles.size();
+            }
+
+            @Override
+            public String getLabel(final int index)
+            {
+                if (index >= 0 && index < styles.size())
+                {
+                    return styles.get(index);
+                }
+                return "";
+            }
+        });
     }
 
     /**
-     * Move the structure down.
+     * Initialise the previous/next and drop down list for schematic.
      */
-    private static void moveDownClicked()
+    private void initSchematicNavigation()
     {
-        Settings.instance.moveTo(new BlockPos(0, -1, 0));
+        registerButton(BUTTON_PREVIOUS_SCHEMATIC_ID, this::previousSchematic);
+        registerButton(BUTTON_NEXT_SCHEMATIC_ID, this::nextSchematic);
+        schematicsDropDownList = findPaneOfTypeByID(DROPDOWN_SCHEMATIC_ID, DropDownList.class);
+        schematicsDropDownList.setHandler((DropDownList.Handler)this);
+        schematicsDropDownList.setDataProvider(new DropDownList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return schematics.size();
+            }
+
+            @Override
+            public String getLabel(final int index)
+            {
+                final String name = schematics.get(index);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "name". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused "name" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1481)